images/cert-manager/config/main.tf

```diff
@@ -0,0 +1,28 @@
+variable "name" {
+  description = "Package name (e.g. cainjector, acmeresolver, controller, webhook)"
+}
+
+variable "suffix" {
+  description = "Package name suffix (e.g. version stream)"
+  default     = ""
+}
+
+variable "extra_packages" {
+  description = "Additional packages to install."
+  type        = list(string)
+  default     = []
+}
+
+module "accts" { source = "../../../tflib/accts" }
+
+output "config" {
+  value = jsonencode({
+    contents = {
+      packages = concat(["cert-manager-${var.name}${var.suffix}"], var.extra_packages)
+    }
+    accounts = module.accts.block
+    entrypoint = {
+      command = "/usr/bin/${var.name}"
+    }
```

Comment:
I tried to do something like this originally with the apko_config, but ran into trouble since most of these fields are marked readonly.

Should we be dropping the read-only designation? (does this even matter w/ tf?)

Reply:
I'm curious what you tried, but `apko_config` takes the string form and _returns_ the parsed/locked config, so it is output-only. However, what `apko_build` accepts is just this structure.

Hopefully that clarifies?